compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -219,9 +220,20 @@ object Completion {
       !sym.is(allOf(Mutable, Accessor)) &&
       (
            (mode.is(Mode.Term) && sym.isTerm)
+        || (mode.is(Mode.StableTerm) && sym.isTerm && validPathSegment(sym))
         || (mode.is(Mode.Type) && sym.isType)
       )
 
+    /** Can this symbol be part of a path? See SLS 3.1 for a definition of a valid path. */
+    private def validPathSegment(sym: Symbol)(implicit ctx: Context): Boolean = {
+      def isRealizable = {
+        val realizability = CheckRealizable.realizability(sym.info)
```

Comment:
I'm afraid that could be a bit slow, I think just checking isStable should be good enough